rel: prep for v0.0.9

arriIsHere opened the pull request:
<!--
Thank you for contributing to the project! 💜
Please see our [OSS process document](https://github.com/honeycombio/home/blob/main/honeycomb-oss-lifecycle-and-practices.md#) to get an idea of how we operate.
-->

## Which problem is this PR solving?

- Closes #<enter issue here>

## Short description of the changes

## How to verify that this has the expected result

---

- [x] CHANGELOG is updated
- [x] README is updated with documentation

Commit:
update version

martin308 reviewed (approved):
Looks like we have also merged two dependabots that should be in the changelog

Commit:
update changelog

arriIsHere commented:
> Looks like we have also merged two dependabots that should be in the changelog

Added the first one. The Otel one we reverted as it includes breaking changes
